packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -478,11 +479,100 @@ function useProcessMedia({
     ]
   );
 
+  /**
+   * Crop video using FFmpeg.
+   *
+   * @param {import('@googleforcreators/media').Resource} resource Resource object.
+   * @param {Object<{newWidth: number, newHeight: number, cropElement: Element}>} cropParams Crop params.
+   */
+  const cropExistingVideo = useCallback(
+    ({ id: elementId, resource: oldResource }, cropParams) => {
+      const { id: resourceId, ...oldResourceWithoutId } = oldResource;
+      const { src: url, mimeType, isOptimized } = oldResource;
+      const { newWidth, newHeight, cropElement } = cropParams;
+
+      const onUploadError = () => {
+        updateExistingElementsByResourceId(resourceId, {
+          height: oldResource.height,
+          width: oldResource.width,
+        });
+      };
+
+      // TODO: Confirm which properties exactly need to be updated.
+      const onUploadProgress = ({ resource }) => {
+        const oldResourceWithId = { ...resource, id: oldResource.id };
+        updateExistingElementsByResourceId(resourceId, {
+          ...oldResourceWithId,
+        });
+      };
+
+      const onUploadSuccess = ({ id, resource }) => {
+        copyResourceData({ oldResource, resource });
+        updateElementById({
+          elementId,
+          properties: {
+            x: cropElement.x < 0 ? 0 : cropElement.x,
+            y:
+              cropElement.y < DANGER_ZONE_HEIGHT
+                ? -DANGER_ZONE_HEIGHT
+                : cropElement.y,
+            width: newWidth,
+            height: newHeight,
+            resource,
+          },
+        });
+
+        // onUploadSuccess is also called with previousResourceId,
+        // for which we don't need to run this.
+        if (id === resource.id) {
+          postProcessingResource(resource);
+        }
+      };
+
+      const process = async () => {
+        let file = false;
+        try {
+          file = await fetchRemoteFile(url, mimeType);
+          await uploadMedia([file], {
+            onUploadSuccess,
+            onUploadProgress,
+            onUploadError,
+            cropVideo: true,
+            additionalData: {
+              original_id: resourceId,
+              cropParams,
+              mediaSource: isOptimized ? 'video-optimization' : 'editor',
+            },
+            originalResourceId: resourceId,
+            resource: {
+              ...oldResourceWithoutId,
+              width: newWidth,
+              height: newHeight,
+            },
+          });
+        } catch (e) {
+          //eslint-disable-next-line no-console -- surface this error
+          console.log(`cropExistingVideo ${e.message}`);
```

Comment:
Let's remove this. We don't console log other errors. Maybe we could use `trackError` and maybe a snackbar message.

Reply:
Added a trackError for now.